lib/stages/visSharedMemWriter.cpp

```diff
@@ -353,9 +350,10 @@ void visSharedMemWriter::main_thread() {
     memcpy(structured_data_addr + 5, &data_size, sizeof(data_size));
 
     // initially set the address records with -1
-    std::vector<char> invalid_vector(ntime * nfreq * access_record_size, invalid);
-    char* tmp = invalid_vector.data();
-    memcpy(access_record_addr, &tmp, sizeof(tmp));
+    // std::fill_n(array, 100, -1);
+    int64_t invalid_vector [ntime * nfreq];
+    std::fill_n(invalid_vector, ntime*nfreq, invalid);
+    memcpy(access_record_addr, invalid_vector, ntime * nfreq * sizeof *invalid_vector);
```

Comment:
typo?

Reply:
No! Apparently `sizeof *invalid_vector` is equivalent to `sizeof(invalid_vector[0])`. But the latter is definitely more clear, so I think I am going to change to the more clear variant.